lib/improver/tests/nowcasting/lightning/test_NowcastLightning.py

```diff
@@ -0,0 +1,946 @@
+        plugin = Plugin()
+        msg = (r"No matching prob\(Ice\) cube for threshold 1.")
+        with self.assertRaisesRegex(ConstraintMismatchError, msg):
+            plugin.apply_ice(self.fg_cube, self.ice_cube)
+
+    def test_missing_threshold_high(self):
+        """Test that the method raises an error if the ice_cube doesn't
+        have a threshold coordinate for 2.0."""
+        self.ice_cube.coord('threshold').points = [0.5, 1., 4.]
+        plugin = Plugin()
+        msg = (r"No matching prob\(Ice\) cube for threshold 2.")
+        with self.assertRaisesRegex(ConstraintMismatchError, msg):
+            plugin.apply_ice(self.fg_cube, self.ice_cube)
+
+    def test_ice_null(self):
+        """Test that small VII probs do not increase lightning risk"""
+        self.ice_cube.data[:, 7, 7] = 0.
+        self.ice_cube.data[0, 7, 7:9] = 0.5
+        self.fg_cube.data[0, 7, 7] = 0.25
+        plugin = Plugin()
+        expected = set_up_cube_with_no_realizations()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 0.25
+        result = plugin.apply_ice(self.fg_cube,
+                                  self.ice_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_ice_zero(self):
+        """Test that zero VII probs do not increase lightning risk"""
+        self.ice_cube.data[:, 7, 7] = 0.
+        self.fg_cube.data[0, 7, 7] = 0.
+        plugin = Plugin()
+        expected = set_up_cube_with_no_realizations()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 0.
+        result = plugin.apply_ice(self.fg_cube,
+                                  self.ice_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_ice_small(self):
+        """Test that small VII probs do increase lightning risk"""
+        self.ice_cube.data[:, 7, 7] = 0.
+        self.ice_cube.data[0, 7, 7] = 0.5
+        self.fg_cube.data[0, 7, 7] = 0.
+        plugin = Plugin()
+        expected = set_up_cube_with_no_realizations()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 0.05
+        result = plugin.apply_ice(self.fg_cube,
+                                  self.ice_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_ice_large(self):
+        """Test that large VII probs do increase lightning risk"""
+        self.ice_cube.data[:, 7, 7] = 1.
+        self.fg_cube.data[0, 7, 7] = 0.
+        plugin = Plugin()
+        expected = set_up_cube_with_no_realizations()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 0.9
+        result = plugin.apply_ice(self.fg_cube,
+                                  self.ice_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_ice_large_long_fc(self):
+        """Test that large VII probs do not increase lightning risk when
+        forecast lead time is large"""
```

Comment:
Is this 3 hours?  Worth saying so as coord units are hidden.

Reply:
Added a comment